RCode/helper_functions/plug_n_play_functions.R

```diff
@@ -29,55 +29,72 @@ plug_n_play_data <- function(start_date, end_date, sites, model_timestep, fill_d
   variable.names.Temp_Quad <- c("tau_add", "beta")
   variable.namesout.Temp_Quad <- c("tau_add", "beta", "mu")
-  
-  data.Logistic <- list(y=y, beta.m=beta.m, beta.v=beta.v, N=length(y),x_ic=x_ic,tau_ic = tau_ic,a_add = a_add,r_add = r_add)
+
+#Logistic
+  data.Logistic <- list(y=y, beta.m=as.vector(c(0,0)), beta.v=solve(diag(1E-03,2)), N=length(y),x_ic=log(0.1),tau_ic = 100,a_add = 0.001,r_add = 0.001)
   variable.names.Logistic <- c("tau_add", "beta")
   variable.namesout.Logistic <- c("tau_add", "beta", "mu")
-  
-  data.Exponential <- list(y=y, beta.m=beta.m, beta.v=beta.v, N=length(y),x_ic=x_ic,tau_ic = tau_ic,a_add = a_add,r_add = r_add)
+
+#Exponential
+  data.Exponential <- list(y=y, beta.m=0, beta.v=1E-03, N=length(y),x_ic=log(0.1),tau_ic = 100,a_add = 0.001,r_add = 0.001)
   variable.names.Exponential <- c("tau_add", "beta")
   variable.namesout.Exponential <- c("tau_add", "beta", "mu")
+
+#RandomYear
+  data.RandomYear <- list(y=y, year_no=year_no, N=length(y),x_ic=log(0.1),tau_ic = 100,a_add = 0.001,r_add = 0.001)
+  variable.names.RandomYear <- c("tau_add", "tau_yr")
+  variable.namesout.RandomYear <- c("tau_add", "tau_yr","yr","mu", "x")
+
+#RandomYearIntercept
+  data.RandomYearIntercept <- list(y=y, year_no=year_no, N=length(y),x_ic=log(0.1),tau_ic = 100,a_add = 0.001,r_add = 0.001, beta.m=0, beta.v=1E-03)
+  variable.names.RandomYearIntercept <- c("tau_add", "tau_yr", "beta")
+  variable.namesout.RandomYearIntercept <- c("tau_add", "tau_yr","yr","mu", "x", "beta")
   
-  data.RandomYear <- list(y=y, beta.m=beta.m, beta.v=beta.v, year_no=year_no, N=length(y),x_ic=x_ic,tau_ic = tau_ic,a_add = a_add,r_add = r_add)
-  variable.names.RandomYear <- c("tau_add", "beta")
-  variable.namesout.RandomYear <- c("tau_add", "tau_yr", "beta", "yr","mu")
-  
-  data.ChangepointTempExp <- list(y=y, beta.m=beta.m, beta.v=beta.v, Temp=Temp, N=length(y),x_ic=x_ic,tau_ic = tau_ic,a_add = a_add,r_add = r_add)
-  variable.names.ChangepointTempExp <- c("tau_add", "beta", "k")
-  variable.namesout.ChangepointTempExp <- c("tau_add", "beta", "k", "mu")
+  #data.ChangepointTempExp <- list(y=y, beta.m=beta.m, beta.v=beta.v, Temp=Temp, N=length(y),x_ic=log(0.1),tau_ic = 100,a_add = 0.001,r_add = 0.001)
+ # variable.names.ChangepointTempExp <- c("tau_add", "beta", "k")
+ # variable.namesout.ChangepointTempExp <- c("tau_add", "beta", "k", "mu")
   
   data = eval(parse(text = paste0('data.', model_name)))
   variable.names = eval(parse(text = paste0('variable.names.', model_name)))
   variable.namesout = eval(parse(text = paste0('variable.namesout.', model_name)))
+  pred_obs= eval(parse(text = paste0('pred_obs.', model_name)))
+  pred = eval(parse(text = paste0('pred.', model_name)))
   
   return(list(data.model = data, variable.names.model = variable.names, variable.namesout.model = variable.namesout)) 
```

Comment:
Also include `pred_obs` and `pred` in the list that's returned?

Reply:
I think my problem was that we had to call this function at the beginning, before the model output to create the preds is even available. But if I create a separate function with just preds like you suggested, I can call that later.